spec/unit/services/index_by_spec.rb

```diff
@@ -0,0 +1,94 @@
+RSpec.describe IndexBy do
+  let(:book) do
+    Product.new(
+      id: 1,
+      name: 'Agile Web Development with Rails 5',
+      price: 2800,
+      vat_category_id: 2
+    )
+  end
+
+  let(:another_book) do
+    Product.new(
+      id: 3,
+      name: 'Web Development with Clojure, Second Edition',
+      price: 2400,
+      vat_category_id: 2
+    )
+  end
+
+  let(:tshirt) do
+    Product.new(
+      id: 6,
+      name: 'Pragmatic T-Shirt',
+      price: 900,
+      vat_category_id: 1
+    )
+  end
```

Comment:
You can try to move them to fixtures, some extra place where you can define them and use in tests